mypy/build.py

```diff
@@ -2237,8 +2317,24 @@ def dispatch(sources: List[BuildSource], manager: BuildManager) -> Graph:
     # just want to load in all of the cache information.
     if manager.use_fine_grained_cache():
         process_fine_grained_cache_graph(graph, manager)
+        # Fine grained protocol dependencies are serialized separately, so we read them
+        # after we loaded cache for whole graph. The `read_protocol_cache` will also validate
+        # the protocol cache against the loaded individual cache files.
+        TypeState.proto_deps = read_protocol_cache(manager, graph)
+        if TypeState.proto_deps is None and manager.stats.get('fresh_trees', 0):
```

Comment:
Maybe write the latter condition could as `manager.stats.get(..., 0) > 0` to be more explicit? Also mention this in the comment, since it looked a bit confusing at first. Also, it could be helpful to document the `stats` attribute (in `BuildManager`, not here) so that it's clear that we rely on the stats for correctness.